### PR TITLE
Do not execute `rtl` function in `styled` components

### PR DESCRIPTION
**Pull request by ciampo**
## Description

The function returned by `rtl()` doesn't need to be invoked inline in `styled` component. This allows Emotion to interpolate that function when props change, thus allowing RTL styles to be more reactive.

## How has this been tested?

Check that the `SelectControl` component behaves correctly in Storybook

## Types of changes

Bug fix

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- N/A I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

**Comment by ciampo** [comment]
@sarayourfriend do you think there's any change to be applied to the `rtl` styles in [`unit-control-styles.js`](https://github.com/WordPress/gutenberg/blob/9d14cdfc47951888aa0fa08ab0afdc680c0316d0/packages/components/src/unit-control/styles/unit-control-styles.js#L92-L93) ?

The function is wrapped in `css()`, which then in turn wrapped in a `styled` component. Not sure if there's any value in refactoring the `rtl` code to be called directly in the `styled` component?

**Comment by sarayourfriend** [comment]
FWIW I believe this is duplicated by https://github.com/WordPress/gutenberg/pull/33696/files#diff-810861ca827286607f443cf248a938682066e055758543043045f63166bc286dL78

But we can merge this and rebase the other PR.

**Review by sarayourfriend — APPROVED** [review]

**Review by sarayourfriend — CHANGES_REQUESTED** [review]
In this case because the return value of `rtl` is being passed as the result of an interpolated props function to the styled components, we do actually need to execute it. It'll get fired on each re-render.

Because these components aren't type checked you won't get an error about this, but you've just changed both the `unitControlBorderRadiusStyles` and `unitControlMarginStyles` signature from:

```ts
function(props): SerializedStyles
```

to

```ts
function(props): () => SerializedStyles
```

This isn't a valid emotion interpolation.

In short, this PR isn't necessary and `rtl` is being used correctly here.

**Comment by ciampo** [comment]
Makes sense, thank you for the explanation.

> Because these components aren't type checked you won't get an error about this

Absolutely, writing components in TypeScript can really help with this sort of changes.

I'm going to close this PR.
